docs/source/docs/level-up/intermediate_skills/index.rst

```diff
@@ -13,6 +13,7 @@ Intermediate Skills
    09_data_filtering
    10_data_exploration
    11_data_generation
+   12_data_pruning
```

Comment:
Since we already have https://github.com/openvinotoolkit/datumaro/blob/a000e3306f021e76c358b79f526321fa93d4065e/docs/source/docs/level-up/advanced_skills/12_project_versioning.rst, if you want to add this to the lesson 12, you have to increment the existing lessons' numbers. Otherwise, append this to the advanced skill section.